### ISSUE TITLE
hw03 ready for grading

### ISSUE THREAD

**Issue by tom-hc-park** [aiod01/STAT545-hw-Hyeongcheol-Park#2]

**Comment by vibudh2209** [comment]
Hi
Overall fantastic work 

- Completed atleast 3 task (Actually all 5!!!)
- Each table accompanied with a proper figure
- Process reported (including difficulties and things which were easy)
- I want to do more (not attempted, its alright as optional)
- Things were not too hard to follow

Few comments:
- Firstly try to provide link to the markdown file in the Readme of hw03 and also would have been great if you mention a brief description about your approach 
- I noticed in min and max gdp you did mutate, because of which you got a lot of rows you could have summarized instead and be done with just 60 rows (one row for each year for each continent)
- Plot for min and max gdp were clear and understandable
- Spread was visualized and tabulated properly
- Here I felt graphs were little difficult to follow due to too many points you could have just taken mean or median for each country and represented it or else just drawn graph for just one single year
- I was not able to understand why were you trying to do linear extrapolation of lifeExp (does not make a lot of sense to me) you could have just taken some kind of weighted mean so that you would have got one value for each year for each continent (this way graphs also wont have too many points and would have been more clear) 
- The number of countries with lifeExp more than a particular value was done very efficiently (I got to learn few things from here), graphs were super clear and overall well done
- For the last part (some special story about any country), things were not very clear to me would have been great it you tried explaining little bit more (you tried and I think the results were interesting so good job here). 
- But you were able to complete 3 and more tasks with good efficiency 

So my final comment is  its a job well done (all my comments are just for improvement as you were able to get the results in most of the parts) and there are surely things which I got know new.

Cheers
Vibudh

**Comment by heathersummers** [comment]
Hi Hyeongcheol,

You did an amazing job with homework #3 and it was very easy to navigate through your repository and assignment. The commenting throughout the assignment made it easy to follow the code, tables, and figures. It was great to see you employed various different types of figures to compliment the various tables. For the first graph of the min and max GDP per capita for each continent per year I would suggest differentiating the min and max values by using different shapes or colours. Also, for the first and third figures in the document there doesn’t need to be a legend with the colours since each continent is labelled in the strip above each panel. For the second task looking at the spread of GDP per capita within the continents it might be good to investigate which country has a much higher GDP per capita than any other country in the 1950s to 1970s. For the figure for task 3 I would suggest adding an alpha value so that it is easier to see the various lines on the graph in particular for Asia and a small stylistic point you might find helpful for future figures is the function `panel.spacing = unit(1.5, “lines”)` in order to increase the spacing between panels so that years in the x-axis are not overlapping onto other panels. I found the last task you completed where you created a new variable called well-being and investigated the country of Kuwait very unique and interesting. I would also suggest having a legend for the last figure, I see there is `legend.position` in the code but there doesn’t appear to be a legend below the figure. Lastly, I would suggest using the `knitr:kable` function for all the tables as the function typically makes it easier for a reviewer to read the table and I would suggest playing around a little more with the figures by adding functions for [plot titles, legends]( https://github.com/hadley/ggplot2-book/blob/master/scales.rmd), [themes](http://ggplot2.tidyverse.org/reference/ggtheme.html), text size, etc. 

Great job on the assignment and all the best with future assignments.
Heather

**Comment by ksedivyhaley** [comment]
**At least three tasks:** Yes (5)
**A table and figure for each task:** Yes
**Comments on tables/figures:**  Yes
**Reflections on Process:** Yes
**Bonus (side-by-side layout, table formatting):** No (basic kable)

**Comments:**

- Creative investigation of the data!
- Boxplots and density plots are usually the best picks for looking at spread.
- Interesting idea trying to scale life expectancy to counteract the increase over time. I don't think it's appropriate here as part of the point of this particular task is to see the increase in life expectancy over time.
- Interesting idea making a “well-being” statistic. However, by subtracting life expectancy you're actually giving countries with a low life expectancy a higher `diff` well-being index.
- You can pipe into kable:
``` r
below %>% 
   group_by(year, continent) %>% 
   summarize(num=n()) %>%
   knitr::kable()
```

Your mark will be distributed later. If you would like more feedback, please feel free to message me on slack.
